### ISSUE TITLE
Performance issues on Totem Destroyer

### ISSUE THREAD

**Issue by Lord-McSweeney** [ruffle-rs/ruffle#9030]
### Describe the bug

On Totem Destroyer (https://www.newgrounds.com/portal/view/447836), the FPS can go below 15, making the game just barely playable. The game runs fine if there are fewer than four blocks left, but if a level has more than ten blocks, the FPS sometimes goes below 8.

### Expected behavior

The game has no performance issues.

### Affected platform

Browser's extension

### Operating system

Linux

### Browser

Chrome 108

**Comment by n0samu** [comment]
Yes, the AVM2 interpreter needs some optimization. Physics simulation with many objects is very slow right now.

**Comment by Lord-McSweeney** [comment]
#9029, #8949, and #8948 all have performance issues with physics simulation.

**Comment by adrian17** [comment]
Do you think it's decent enough now?
